docs/guide/en/widget-configuring.md

```diff
@@ -44,3 +57,67 @@ parameter:
 
 For a description of the configuration syntax, see the
 [Yii Definitions](https://github.com/yiisoft/definitions#arraydefinition) package documentation.
+
+## Widget factory configuration
+
+Widget factory could be used to configure widgets. It is handy to use it to set global defaults.
+
+### With definitions
+
+You can use definitions:
+
+```php
+\Yiisoft\Widget\WidgetFactory::initialize(
+    /** @var \Psr\Container\ContainerInterface $container */
+    $container,
+    definitions: [
+        MyWidget::class => [
+            '__construct()' => [
+                'name' => 'Base',
+            ],
+        ],
+    ],
+);
+```
+
+For a description of the configuration syntax, see the
+[Yii Definitions](https://github.com/yiisoft/definitions#arraydefinition) package documentation.
+
+### With themes
+
+Themes could be used in case you want to have multiple configuration sets and the ability to switch from one to another.
+Theme configuration is named and merged with default configuration. Themes are defined in `WidgetFactory::initialize()`.
+To apply the theme automatically, specify it as default theme.
+
+```php
+\Yiisoft\Widget\WidgetFactory::initialize(
+    /** @var \Psr\Container\ContainerInterface $container */
+    $container,
+    themes: [
+        'red-alert' => [
+            MyWidget::class => [
+                '__construct()' => [
+                    'color' => 'red',
+                ],
+            ],
+        ],
+        'black' => [
+            MyWidget::class => [
+                '__construct()' => [
+                    'color' => 'black',
+                ],
+            ],
+        ],
+    ],
+    defaultTheme: 'black',
+);
+```
+
+## With a theme specified at the `widget()` method call.
```

Comment:
Need info about extra configuration from widget itself.

Reply:
Do you mean the fact that theme config is merged with widget call level config?

Reply:
No. I means such cases as field in `yiisoft/form`: https://github.com/yiisoft/form/blob/e0daafe346b4ac20ed4893451e9a361c47bc9cf0/src/Field/Base/BaseField.php#L150

We have four levels:

1) configuration passed to the `widget()` method call;
2) extra configuration from the widget itself by specified theme (see case in `yiisoft/form`);
3) configuration defined in widget factory themes:
4) configuration defined in widget factory definitions.